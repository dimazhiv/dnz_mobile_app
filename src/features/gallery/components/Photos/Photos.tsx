import React from 'react';
import { FlatList, StyleSheet, View } from 'react-native';
import PhotoItem from './PhotoItem/PhotoItemContainer';

export type PhotosProps = {
  uris: string[];
};

const renderItem = ({ item }: { item: string }) => <PhotoItem uri={item} />;

const renderSeparator = () => <View style={styles.separator} />;

function Photos({ uris }: PhotosProps) {
  return (
    <FlatList
      contentContainerStyle={styles.flatList}
      ItemSeparatorComponent={renderSeparator}
      data={uris}
      renderItem={renderItem}
      showsVerticalScrollIndicator={false}
    />
  );
}

const styles = StyleSheet.create({
  flatList: {
    paddingTop: 10,
    paddingBottom: 20
  },
  separator: {
    flex: 1,
    paddingTop: 20
  }
});

export default Photos;
