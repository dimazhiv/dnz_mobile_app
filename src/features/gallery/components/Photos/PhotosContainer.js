import { connect } from 'react-redux';
import { getPhotosUri } from '../../selectors';
import Photos from './Photos';

function mapStateToProps(state) {
  return {
    uris: getPhotosUri(state)
  };
}

export default connect(mapStateToProps, null)(Photos);
