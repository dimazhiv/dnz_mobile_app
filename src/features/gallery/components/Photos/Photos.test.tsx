import React from 'react';
import { shallow } from 'enzyme';
import Photos from './Photos';

describe('Snapshot testing Photos component', () => {
  test('correct component view', () => {
    const tree = shallow(
      <Photos
        uris={[
          'https://farm9.staticflickr.com/8187/8432423659_dd1b834ec5.jpg',
          'https://farm9.staticflickr.com/8450/8047948330_b0e55d455f.jpg'
        ]}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
