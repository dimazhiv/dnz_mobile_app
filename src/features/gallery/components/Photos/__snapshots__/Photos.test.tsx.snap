// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing Photos component correct component view 1`] = `
<FlatList
  ItemSeparatorComponent={[Function]}
  contentContainerStyle={
    {
      "paddingBottom": 20,
      "paddingTop": 10,
    }
  }
  data={
    [
      "https://farm9.staticflickr.com/8187/8432423659_dd1b834ec5.jpg",
      "https://farm9.staticflickr.com/8450/8047948330_b0e55d455f.jpg",
    ]
  }
  renderItem={[Function]}
  showsVerticalScrollIndicator={false}
/>
`;
